app/models/assignment_generator.rb

```diff
@@ -0,0 +1,308 @@
+# A challenge assignment generator that uses Hopcroft-Karp to try to get a
+# maximum matching (thus maximizing the number of complete assignments, and
+# minimizing the number of assignments with no recipient or no giver). Sorts
+# the potential matches (using the default sort, which relies on the fields
```

Comment:
Can you add a comment in PotentialMatch's `<=>` that the assignment generator depends on its logic?